source/vibe/http/client.d

```diff
@@ -194,20 +197,99 @@ auto connectHTTP(string host, ushort port = 0, bool use_tls = false, HTTPClientS
 
 /**
-	Defines an HTTP/HTTPS proxy request or a connection timeout for an HTTPClient.
+	Defines the options used in an HTTPClient for defining its requests.
+
+	A new connection will be opened in requestHTTP for each different HTTPClientSettings.
 */
-class HTTPClientSettings {
+final class HTTPClientSettings {
+	/** Sets the URL of a proxy to use for requests.
+
+		If an HTTP proxy is used, the URL must be provided. It will be resolved,
+		and the `HTTPClient` will throw if it cannot connect to it.
+	*/
 	URL proxyURL;
-	Duration defaultKeepAliveTimeout = 10.seconds;
+
+	/** Default maximum amount of time to keep a connection open after a request.
+
+		Maximum amount of time the client should wait for the next request when
+		there are none active. Note that the server will usually report its own
+		keep-alive timeout, which will then be used instead.
+
+		This setting is obeyed for HTTP/1.1 and HTTP/2 connections.
+	*/
+	Duration maxKeepAliveTimeout = 10.seconds; 
+
+	/** Sets the user agent string to report in requests.
+
+		If left empty, the default vibe.d user agent string will be used. Use
+		`HTTPClient.setUserAgentString` to set a different default user agent
+		globally.
+	*/
+	string userAgent;
+
+	/// General option flags
+	HTTPClientOptions options = HTTPClientOption.defaults;
+
+	/// All cookies will be processed from and to the cookie jar if specified
+	CookieStore cookieJar;
+
+	/** If set, sends ping frames at regular intervals to avoid peer
+		inactivity timeout.
+
+		Note that this field only has an effect for HTTP/2 connections.
+	*/
+	Duration pingInterval; 
+	
+	/** Maxiumum time to wait when reading or writing data.
+
+		This setting only has an effect for HTTP/2 connections. When a read or
+		write operation takes longer than set here, the connection gets dropped.
+	*/
+	Duration connectionTimeout = 5.minutes;
+	
+	/// Custom TLS context for this client
+	TLSContext tlsContext;
+}
+
+
+/** Option flags used to configure the HTTP client.
+*/
+enum HTTPClientOption {
+	/** Uses HTTP/2 requests without prior checks.
+
+		If enabled, the client will always send a client preface without trying
+		upgrade or checking the ALPN value.
+	*/
+	forceHTTP2,
+	
+	/** Disables HTTP/2 and performs only HTTP/1.x requests.
+	
+		If left enabled, HTTP/2 upgrade will take place and after a successful
+		connection, all further (concurrent or sequential) requests will be
+		multiplexed over the same TCP Connection until it is timed out through
+		keep-alive or inactivity.
+	*/
+	disableHTTP2,
+
+	/// Will not try to upgrade the connection on non-TLS connections.
+	onlyEncryptedHTTP2,
+
+	/// Default set of flags - enables normal HTTP/2 upgrades on TLS connections.
+	defaults = onlyEncryptedHTTP2,
 }
 
+/*static if (__VERSION__ >= 2067) {
+	import std.typecons : BitFlags;
+	alias HTTPClientOptions = BitFlags!HTTPClientOption;
+} else*/ alias HTTPClientOptions = HTTPClientOption;
+
```

Comment:
`BitFlags` was just added in 2.067 if I remember right, but I'm not sure why this ended up commented out. Personally I'd be in favor of using an `EnumSet`, though, `BitFlags` seems to be too much of a leaky abstraction.